homeassistant/components/hassio/update.py

```diff
@@ -0,0 +1,288 @@
+"""Update platform for Supervisor."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.update import (
+    UpdateEntity,
+    UpdateEntityDescription,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ICON, ATTR_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import (
+    ADDONS_COORDINATOR,
+    async_update_addon,
+    async_update_core,
+    async_update_os,
+    async_update_supervisor,
+)
+from .const import (
+    ATTR_CHANGELOG,
+    ATTR_VERSION,
+    ATTR_VERSION_LATEST,
+    DATA_KEY_ADDONS,
+    DATA_KEY_OS,
+    DATA_KEY_SUPERVISOR,
+)
+from .entity import (
+    HassioAddonEntity,
+    HassioCoreEntity,
+    HassioOSEntity,
+    HassioSupervisorEntity,
+)
+from .handler import HassioAPIError
+
+ENTITY_DESCRIPTION = UpdateEntityDescription(
+    name="Update",
+    key=ATTR_VERSION_LATEST,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Supervisor update based on a config entry."""
+    coordinator = hass.data[ADDONS_COORDINATOR]
+
+    entities = [
+        SupervisorSupervisorUpdateEntity(
+            coordinator=coordinator,
+            entity_description=ENTITY_DESCRIPTION,
+        ),
+        SupervisorCoreUpdateEntity(
+            coordinator=coordinator,
+            entity_description=ENTITY_DESCRIPTION,
+        ),
+    ]
+
+    for addon in coordinator.data[DATA_KEY_ADDONS].values():
+        entities.append(
+            SupervisorAddonUpdateEntity(
+                addon=addon,
+                coordinator=coordinator,
+                entity_description=ENTITY_DESCRIPTION,
+            )
+        )
+
+    if coordinator.is_hass_os:
+        entities.append(
+            SupervisorOSUpdateEntity(
+                coordinator=coordinator,
+                entity_description=ENTITY_DESCRIPTION,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class SupervisorAddonUpdateEntity(HassioAddonEntity, UpdateEntity):
+    """Update entity to handle updates for the Supervisor add-ons."""
+
+    _attr_supported_features = UpdateEntityFeature.INSTALL | UpdateEntityFeature.BACKUP
+
+    @property
+    def title(self) -> str | None:
+        """Return the title of the update."""
+        if not self.available:
+            return None
```

Comment:
We do not need to check this, if it is unavailable, the state will not be updated/called.

```suggestion
```